core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -0,0 +1,529 @@
+        }
+      }
+      topicInfoList.add(topicInfo);
+    }
+
+    if (topicOverview.getTopicIdForDocumentation() == null) {
+      topicOverview.setTopicDocumentation(topics.get(0).getDocumentation());
+      topicOverview.setTopicIdForDocumentation(topics.get(0).getTopicid());
+    }
+
+    topicOverview.setTopicHistoryList(topicHistoryList);
+  }
+
+  private List<AclInfo> applyFiltersAclsForSOT(
+      Integer loggedInUserTeam, List<Acl> aclsFromSOT, int tenantId) {
+
+    List<AclInfo> aclList = new ArrayList<>();
+    AclInfo mp;
+
+    for (Acl aclSotItem : aclsFromSOT) {
+      mp = new AclInfo();
+      mp.setEnvironment(aclSotItem.getEnvironment());
+      mp.setEnvironmentName(getEnvDetails(aclSotItem.getEnvironment(), tenantId).getName());
+      mp.setTopicname(aclSotItem.getTopicname());
+      mp.setAcl_ip(aclSotItem.getAclip());
+      mp.setAcl_ssl(aclSotItem.getAclssl());
+      mp.setTransactionalId(aclSotItem.getTransactionalId());
+      mp.setTeamname(manageDatabase.getTeamNameFromTeamId(tenantId, aclSotItem.getTeamId()));
+      mp.setConsumergroup(aclSotItem.getConsumergroup());
+      mp.setTopictype(aclSotItem.getTopictype());
+      mp.setAclPatternType(aclSotItem.getAclPatternType());
+      mp.setReq_no(aclSotItem.getReq_no() + "");
+      if (aclSotItem.getTeamId() != null && aclSotItem.getTeamId().equals(loggedInUserTeam))
+        mp.setShowDeleteAcl(true);
+
+      if (aclSotItem.getAclip() != null || aclSotItem.getAclssl() != null) aclList.add(mp);
+    }
+    return aclList;
+  }
+
+  private List<Acl> getAclsFromSOT(
+      String env, String topicNameSearch, boolean regex, int tenantId) {
+    List<Acl> aclsFromSOT;
+    if (!regex) {
+      aclsFromSOT =
+          manageDatabase.getHandleDbRequests().getSyncAcls(env, topicNameSearch, tenantId);
+    } else {
+      aclsFromSOT = manageDatabase.getHandleDbRequests().getSyncAcls(env, tenantId);
+      List<Acl> topicFilteredList = aclsFromSOT;
+      // Filter topics on topic name for search
+      if (topicNameSearch != null && topicNameSearch.length() > 0) {
+        final String topicSearchFilter = topicNameSearch;
+        topicFilteredList =
+            aclsFromSOT.stream()
+                .filter(acl -> acl.getTopicname().contains(topicSearchFilter))
+                .collect(Collectors.toList());
+      }
+      aclsFromSOT = topicFilteredList;
+    }
+
+    return aclsFromSOT;
+  }
+
+  private Map<String, String> getTopicPromotionEnv(
+      String topicSearch, List<Topic> topics, int tenantId) {
+    Map<String, String> hashMap = new HashMap<>();
+    try {
+      if (topics == null) {
+        topics = manageDatabase.getHandleDbRequests().getTopics(topicSearch, tenantId);
+      }
+
+      hashMap.put("topicName", topicSearch);
+
+      if (topics != null && topics.size() > 0) {
+        List<String> envList = new ArrayList<>();
+        topics.forEach(topic -> envList.add(topic.getEnvironment()));
```

Comment:
```suggestion
        List<String> envList = new ArrayList<>();
        topics.forEach(topic -> envList.add(topic.getEnvironment()));
```
In this case list will be created of better suitable size

Reply:
Looks good. updated.